project/apps/web/app/profile/page.tsx

```diff
@@ -1,62 +1,55 @@
 'use client';
 
-import { UpdateUserDto, updateUserSchema } from '@repo/dtos/users';
-import { useMutation, useQueryClient } from '@tanstack/react-query';
-import { Suspense, useState } from 'react';
+import { UpdateUserDto } from '@repo/dtos/users';
+import { useMutation } from '@tanstack/react-query';
+import { Suspense } from 'react';
 
 import { ActionModals } from '@/components/profile/ActionModals';
+import ProfileDetails from '@/components/profile/ProfileDetails';
 import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
 import { Button } from '@/components/ui/button';
-import {
-  Form,
-  FormControl,
-  FormField,
-  FormItem,
-  FormLabel,
-} from '@/components/ui/form';
-import { Input } from '@/components/ui/input';
-import { QUERY_KEYS } from '@/constants/queryKeys';
-import {
-  ProfileStateProvider,
-  useProfileState,
-} from '@/contexts/ProfileStateContext';
 import { useToast } from '@/hooks/use-toast';
 import { updateUser } from '@/lib/api/users';
-import { useZodForm } from '@/lib/form';
 import { useAuthStore } from '@/stores/useAuthStore';
+import { useProfileStore } from '@/stores/useProfileStore';
 
-import ProfileSkeleton from './components/ProfileSkeleton';
+import ProfileSkeleton from '../../components/profile/ProfileSkeleton';
 
 const ProfilePageContent = () => {
   const user = useAuthStore.use.user();
-  const form = useZodForm({ schema: updateUserSchema });
-  const [isEditingUsername, setIsEditingUsername] = useState(false);
-  const [isEditingEmail, setIsEditingEmail] = useState(false);
+  if (!user) return;
+  const fetchUser = useAuthStore.use.fetchUser();
+  const { toast } = useToast();
   const {
+    isEditingUsername,
+    setIsEditingUsername,
+    isEditingEmail,
+    setIsEditingEmail,
     confirmLoading,
     setConfirmLoading,
     isChangePasswordModalOpen,
     isDeleteModalOpen,
     setChangePasswordModalOpen,
     setDeleteModalOpen,
```

Comment:
comment: avoid destructuring when using stores unless this component is expected to re-render when anything in the store changes